modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/ColocationGroup.java

```diff
@@ -40,11 +42,27 @@ public class ColocationGroup implements Serializable {
 
     private final List<NodeWithConsistencyToken> assignments;
 
+    private final Map<String, List<PartitionWithConsistencyToken>> partitionsPerNode;
+
     /** Constructor. */
     public ColocationGroup(List<Long> sourceIds, List<String> nodeNames, List<NodeWithConsistencyToken> assignments) {
         this.sourceIds = Objects.requireNonNull(sourceIds, "sourceIds");
         this.nodeNames = Objects.requireNonNull(nodeNames, "nodeNames");
         this.assignments = Objects.requireNonNull(assignments, "assignments");
+        this.partitionsPerNode = null;
+    }
+
+    /** Constructor for colocation group with applied partition pruning. */
```

Comment:
This is stored separately because  `IgniteTrimExchange` creates a destination that expects the number of assignments as stored for a table (see DestinationFactory).